src/Elastic.Apm/Helpers/SystemInfoHelper.cs

```diff
@@ -83,26 +74,20 @@ private Container ParseContainerId(string line)
 
 			// If the line matched the one of the kubernetes patterns, we assume that the last part is always the container ID.
 			// Otherwise we validate that it is a 64-length hex string
-			if (!string.IsNullOrWhiteSpace(kubernetesPodUid) || _containerUidRegex.IsMatch(idPart) || _shortenedUuidRegex.IsMatch(idPart))
-				return new Container { Id = idPart };
-
-			_logger.Info()?.Log("Could not parse container ID from '/proc/self/cgroup' line: {line}", line);
-			return null;
+			if (system.Kubernetes != null || _containerUidRegex.IsMatch(idPart) || _shortenedUuidRegex.IsMatch(idPart))
+				system.Container = new Container { Id = idPart };
+			else
+				_logger.Info()?.Log("Could not parse container ID from '/proc/self/cgroup' line: {line}", line);
 		}
 
 		internal Api.System ParseSystemInfo(string hostName)
```

Comment:
This is really a nit, but maybe call it `GetSystemInfo`? 

Reason I mention is that the signatures changed, all `ParseX` methods are now in `void ParseX(variableToFill)` format in this file, except this one. With that it'd also match the signiture/naming of the `GetHostName` method.